sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataTypeParser.scala

```diff
@@ -67,6 +68,11 @@ private[sql] trait DataTypeParser extends StandardTokenParsers {
       case tpe => ArrayType(tpe)
     }
 
+  protected lazy val standardArrayType: PackratParser[DataType] =
+    (standardArrayType | primitiveType) <~ "(?i)array".r <~ ("[" ~ numericLit ~ "]").? ^^ {
```

Comment:
Having the capacity part in the syntax but ignoring them seems confusing. I'd suggest removing the capacity part. If we do decide to add this feature, we can add it back again.